### ISSUE TITLE
Error in extracting agent from GeoSpace.agents in tutorial using Spyder

### ISSUE THREAD

**Issue by whitbern** [projectmesa/mesa-geo#193]
this is an attachment of the script [mesa_geo_tutorial.txt](https://github.com/projectmesa/mesa-geo/files/14596526/mesa_geo_tutorial.txt) 
this is a screenshot of the error [error_mesa_geo_tutorial](https://github.com/projectmesa/mesa-geo/assets/53237071/0d88755c-1b6e-48ae-8407-48483c5d32b0)

Good evening, currently, I am doing the tutorial for Mesa-geo using Spyder. When I try to extract the agents using this line:

agent = m.space.agents[0]

I get a TypeError stating that: 'type' object is not subscriptable. Before that, I get a "Coordinate = tuple[int, int]." However, when I look at the mesa.space module, they mention lists? 

I would expect for this just to give me the first agent from the listed url.

Thankfully this works just fine in Jupyter Notebook. Can you all help me figure out why this is not working in Spyder?
Thanks!

**Comment by wang-boyu** [comment]
Thanks for sharing the code and screenshot. The error appears to be from type hints. Can I ask which version of Python you were using?

Mesa & Mesa-Geo require Python 3.9 and above. I couldn't reproduce the error with Python 3.10. Perhaps try to recreate a clean virtual environment with Python 3.9 or above, install Mesa-Geo and try again?

**Comment by EwoutH** [comment]
This is indeed an error caused by new syntax introduced in Python 3.9. Upgrading should resolve the problem.

@wang-boyu the more interesting problem is how users manage to install Mesa-geo with incompatible Python versions. Have we set our Python requirements correctly in this repo? https://packaging.python.org/en/latest/guides/dropping-older-python-versions/#specify-the-version-ranges-for-supported-python-distributions

**Comment by wang-boyu** [comment]
Python requirement is specifies in the `setup.cfg` file: https://github.com/projectmesa/mesa-geo/blob/b717971075a99ed18cd06126183c75407baa8dcb/setup.cfg#L29 I suppose it should work as expected...?

**Comment by EwoutH** [comment]
Yes, this is indeed the case and it's configured correctly: https://github.com/EwoutH/mesa/actions/runs/8379466683

@whitbern which version of mesa-geo are you using and which Python version? How did you install mesa-geo?

**Comment by whitbern** [comment]
Hello--I have Python 3.11.5 I have the latest version of mesa-geo: 0.7.0. I installed mesa-geo using these commands for Windows in the Anaconda Prompt: 

conda install fiona pyproj rtree shapely
pip install mesa-geo

**Comment by EwoutH** [comment]
Right, I read the error wrongly. Seems like an issue in the GeoSpace or the mesa space itself.

1. Which mesa version are you using? 
2. Could you print `m.space.agents` and its type.
3. Could you update to the latest versions of both mesa and mesa-geo and see if this error still persists?
